app/models/invoice/data_presenter.rb

```diff
@@ -68,19 +68,30 @@ def checkout_adjustments(exclude: [], reject_zero_amount: true)
     end
 
     def display_checkout_taxes_hash
-      totals = OrderTaxAdjustmentsFetcher.new(nil).totals(all_tax_adjustments)
-
-      totals.map do |tax_rate, tax_amount|
+      tax_adjustment_totals.map do |tax_rate_id, tax_amount|
+        tax_rate = tax_rate_by_id[tax_rate_id]
         {
-          amount: Spree::Money.new(tax_amount, currency: order.currency),
+          amount: Spree::Money.new(tax_amount, currency: currency),
           percentage: number_to_percentage(tax_rate.amount * 100, precision: 1),
           rate_amount: tax_rate.amount,
         }
       end.sort_by { |tax| tax[:rate_amount] }
     end
 
     def all_tax_adjustments
-      all_eligible_adjustments.select { |a| a.originator_type == 'Spree::TaxRate' }
+      all_eligible_adjustments.select { |a| a.originator.type == 'Spree::TaxRate' }
+    end
+
+    def tax_adjustment_totals
+      all_tax_adjustments.each_with_object(Hash.new(0)) do |adjustment, totals|
```

Comment:
Could you use `{}` instead of `Hash.new(0)` like below ? or vice versa, it would be better if they were consistent.